### PR TITLE
Make Docker build more robust and other improvements for Dockerfile

### PR DESCRIPTION
**Pull request by stweil**
- Allow some failures for BNB downloads.
- Add support for podman.
- Update to Debian stable.
- Upgrade jQuery and clipboard.js.
- ... and two more small improvements.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
-FROM php:7.3-apache-buster
+FROM docker.io/php:8.2-apache-bullseye
 
 ENV JQUERY 3.6.0
 ENV CLIPBOARD 2.0.8
 
-RUN apt-get update \
-    && apt-get install --no-install-recommends -y yaz libyaz-dev wget unzip python3-bs4 python3-requests \
+RUN apt-get update && apt-get upgrade -y
+RUN apt-get install --no-install-recommends -y yaz libyaz-dev wget unzip python3-bs4 python3-requests \
```
Comment (zuphilip) [inline Dockerfile:14]: It is recommened to "always combine RUN apt-get update with apt-get install in the same RUN statement", see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get

File: Dockerfile [hunk 2]
```diff
@@ -5,13 +5,13 @@
...
 
-FROM php:7.3-apache-buster
+FROM docker.io/php:8.2-apache-bullseye
 
 ENV JQUERY 3.6.0
 ENV CLIPBOARD 2.0.8
 
-RUN apt-get update \
-    && apt-get install --no-install-recommends -y yaz libyaz-dev wget unzip python3-bs4 python3-requests \
+RUN apt-get update && apt-get upgrade -y
```
Comment (zuphilip) [inline Dockerfile:13]: Is an upgrade really needed? - I think that the recommendations were more going in the direction that one should not do such an upgrade when using an official Docker image which is by itself maintained. However, now it seems that one can find pros and cons for doing this in a docker container or not.

Comment (stweil) [reply Dockerfile]: I removed the commit which changed that part now, because I can no longer reproduce the problem which I saw yesterday.

**Comment by stweil** [comment]
@zuphilip, I removed the commit which changed the `apt-get` commands from the pull request.

File: bnb/getBNBData [hunk 3]
```diff
@@ -88,7 +88,7 @@ def main():
             retcode += 1
             continue
         if r.headers['content-type'] != 'application/x-zip-compressed':
-            print(f"Unexpected content type {r.headers['content-type']} for {zipurl}")
+            print(f"Warning: Unexpected content type {r.headers['content-type']} for {zipurl}")
             retcode += 1
```
Comment (zuphilip) [inline bnb/getBNBData:92]: Is this change needed? - The return code is still >0 and therefore I am not sure that it makes sense to change the wording here...

**Comment by zuphilip** [comment]
@stweil Thank you very much. I have one question/suggestion.

General remark: We can change the Dockerfile to use a new PHP version and library version. At some point we should then also update the information in the INSTALL.md and make this consistently.

File: bnb/getBNBData (hunk 3 above)
Comment (stweil) [reply bnb/getBNBData:92]: This change is for humans. "Unexpected content type" sounds like an error for me, "Warning: Unexpected content type" makes it clear that it's only a warning, but the result can still be valid.

Tagging log messages as info, warning and error is common practice.
